### PR TITLE
fix: Fix #24320: Prevent editing of safe-transfer-from transactions

### LINKED ISSUE #24320: [Bug]: Editing a safeTransferFrom transaction request converts it to a transferFrom
### Describe the bug

`safeTransferFrom` transaction requests are incorrectly converted to `transferFrom` transaction if users navigate to edit and then back to the transaction. 

Introduced on v11.9.0 with https://github.com/MetaMask/metamask-extension/pull/22228/

Expected to be fixed on v11.15.1 with https://github.com/MetaMask/metamask-extension/pull/24319

### Expected behavior

Transaction requests are not converted from safeTransferFrom to transferFrom.

### Screenshots/Recordings

https://github.com/MetaMask/metamask-extension/assets/34968356/e40cdcf3-8ac7-4f7d-aa82-1200311d356c

### Steps to reproduce

1. On [MM test dapp](https://metamask.github.io/test-dapp/), connect to an account on Sepolia
2. Click on NFTs Deploy
3. Click on Mint
4. Go to your NFT contract on the Sepolia block explorer
5. Go to write Contract tab
6. Connect your wallet
7. Scroll to safeTransferFrom and fill in the info
8. Click write
9. On the MM popup, notice `safeTransferFrom` as the method
10. Click on edit on the confirmation
11. Click on Next button
12. Notice `transferFrom` as the method on the confirmation

Alternatively, you can also skip steps 1 to 4 and use [this link](https://sepolia.etherscan.io/address/0x07B6C897a68D53948422B534935C4f40c5F25DaE#writeContract). The only difference is that as you don't own the NFT you'll see warnings for gas estimation failures.

### Error messages or log output

_No response_

### Version

11.9.0

### Build type

None

### Browser

Brave

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

Ensures that ERC1155s sends are no longer editable

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24319?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24320

## **Manual testing steps**

See issue for STRs

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/components/confirm-page-container/confirm-page-container.component.js [hunk 1]
```diff
@@ -213,7 +214,12 @@ const ConfirmPageContainer = (props) => {
           />
         ) : (
           <ConfirmPageContainerHeader
-            showEdit={showEdit}
+            showEdit={
+              origin === ORIGIN_METAMASK ||
+              currentTransaction.type === TransactionType.simpleSend
+                ? showEdit
```
Comment (Gudahtt) [inline ui/pages/confirmations/components/confirm-page-container/confirm-page-container.component.js:220]: We should follow up in a later PR to simplify this logic a bit (I think we can inline the remaining useful parts of `showEdit` here). There might be more dead code to remove as well.

Comment (Gudahtt) [reply ui/pages/confirmations/components/confirm-page-container/confirm-page-container.component.js:220]: We should also follow up with e2e test coverage for any impacted types of transactions we can think of

**Review by Gudahtt — DISMISSED** [review]
LGTM!

File: ui/pages/confirmations/components/confirm-page-container/confirm-page-container.component.js [hunk 2]
```diff
@@ -213,7 +214,12 @@ const ConfirmPageContainer = (props) => {
           />
         ) : (
           <ConfirmPageContainerHeader
-            showEdit={showEdit}
+            showEdit={
+              origin === ORIGIN_METAMASK ||
+              currentTransaction.type === TransactionType.simpleSend
```
Comment (Gudahtt) [inline ui/pages/confirmations/components/confirm-page-container/confirm-page-container.component.js:219]: If we wanted to continue allowing token transfer edits:
```suggestion
              [TransactionType.simpleSend, TransactionType.tokenMethodTransfer].includes(currentTransaction.type)
```

**Review by cryptotavares — DISMISSED** [review]

**Review by danjm — APPROVED** [review]

**Review by Gudahtt — APPROVED** [review]

File: ui/pages/confirmations/confirm-transaction/confirm-token-transaction-switch.js [hunk 3]
```diff
@@ -144,14 +144,6 @@ export default function ConfirmTokenTransactionSwitch({ transaction }) {
             decimals={decimals}
             image={tokenImage}
             tokenAddress={tokenAddress}
-            onEdit={async ({ txData }) => {
```
Comment (Gudahtt) [inline ui/pages/confirmations/confirm-transaction/confirm-token-transaction-switch.js:147]: Actually wait, it looks like this component assumes that `onEdit` is always passed in

**Review by Gudahtt — CHANGES_REQUESTED** [review]

File: ui/pages/confirmations/confirm-transaction/confirm-token-transaction-switch.js (hunk 3 above)
Comment (Gudahtt) [reply ui/pages/confirmations/confirm-transaction/confirm-token-transaction-switch.js:147]: Following the trail here: this is a non-required prop, passed into `confirm-token-transaction-base`. That component passes it to `confirm-transaction-base` (passed through the container untouched, to the component module).

It's called here: https://github.com/MetaMask/metamask-extension/blob/4b543cd1fc843547e10096a3591f28036aa303b9/ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js#L617

We try to call this in `handleEdit` without checking that it's defined.

Comment (Gudahtt) [reply ui/pages/confirmations/confirm-transaction/confirm-token-transaction-switch.js:147]: Oh but wait, we also set `showEdit` to `false` when `onEdit` is false, here: https://github.com/MetaMask/metamask-extension/blob/4b543cd1fc843547e10096a3591f28036aa303b9/ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js#L1099

Super confusing how this is setup, but never mind, this should work as-is.

**Review by Gudahtt — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Pull request merged**